chainer/backends/_chainerx.py

```diff
@@ -37,9 +37,10 @@ def from_array(array):
     def from_fallback_device(device):
         # TODO(niboshi): Write unit test
         assert isinstance(device, _backend.Device)
-        if isinstance(device, _cpu.CpuDevice):
+        assert not isinstance(device, intel64.Intel64Device)
+        if device.xp is numpy:
             return ChainerxDevice(chainerx.get_device('native', 0))
-        if isinstance(device, cuda.GpuDevice):
+        if device.xp is cuda.cupy:
```

Comment:
The previous logic is preferable.
* This method is irrelevant to the existence of `Intel64Device`. It should not appear in the code.
* In this implementation it causes assertion error instead of `RuntimeError` for an `Intel64Device` input.

Reply:
Reverted change.